Content.Client/GameObjects/Components/PipeDispenser/PipeDispenserVisualizer.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using Robust.Client.Animations;
+using Robust.Client.GameObjects;
+using Robust.Client.GameObjects.Components.Animations;
+using Robust.Client.Interfaces.GameObjects.Components;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Utility;
+using YamlDotNet.RepresentationModel;
+using static Content.Shared.GameObjects.Components.SharedPipeDispenserComponent;
+
+namespace Content.Client.GameObjects.Components.VendingMachines
+{
+    [UsedImplicitly]
+    public sealed class PipeDispenserVisualizer : AppearanceVisualizer
+    {
+        private Dictionary<string, bool> _baseStates;
+
+        private static readonly Dictionary<string, PipeDispenserVisualLayers> LayerMap =
+            new Dictionary<string, PipeDispenserVisualLayers>
+            {
+                {"off", PipeDispenserVisualLayers.Unlit},
+                {"screen", PipeDispenserVisualLayers.Screen},
+                {"normal", PipeDispenserVisualLayers.Base},
+                {"normal-unshaded", PipeDispenserVisualLayers.BaseUnshaded},
+                {"eject", PipeDispenserVisualLayers.Base},
+                {"eject-unshaded", PipeDispenserVisualLayers.BaseUnshaded},
+                {"broken", PipeDispenserVisualLayers.Unlit},
+            };
+
+        private Dictionary<string, Animation> _animations = new Dictionary<string, Animation>();
+
+        public override void LoadData(YamlMappingNode node)
+        {
+            base.LoadData(node);
+
+            _baseStates = new Dictionary<string, bool>
```

Comment:
I also don't like this whole saving if the states exist. Just do a LayerMap.TryGetValue if they are asked